tests/manage/test_add_mds_to_cluster.py

```diff
@@ -69,25 +29,45 @@ def verify_fs_exist(pod_count):
     return False
 
 
+@pytest.fixture(scope="class")
+def cephfs():
+    cephfs = ocp.OCP(kind=constants.CEPHFILESYSTEM)
+    return cephfs
+
+
+@pytest.fixture(scope="class")
```

Comment:
Because we only have one test method in this class I'm not entirely sure what the correct scope should be. Setting it to ``class`` would mean that each test method within the class using the fixture would get the same copy of ``fs_data``. If we set it to ``function`` then each test method would get a fresh copy. The original teardown was resetting a value on ``fs_data`` to the original value. If we set the scope to ``function`` that wouldn't be necessary.

It seems to me that if we were gonna move these functions to conftest.py we'd want the scope set to ``function`` so each test method could be isolated?

Reply:
@andrewschoen I agree with function scope, especially if they move to `conftest.py`. Unless it's necessary to share state between tests, function scope makes sense. Creating new objects is cheap and having state shared automatically can cause unintended issues if it isn't accounted for.

Reply:
for some objects class makes sense and for some function, I think this is all test dependent, at the conftest.py we should default to class level and individual tests can create their own function level scopes

Reply:
@vasukulkarni I think it would be better for us to align with what pytest itself defaults to which is function scope. If a fixture is explicitly designed to happen once per test class then we can set class scope, otherwise we can assume that a fixture is creating a resource for one test only, and tests are receiving resources that were not possibly altered by other tests.

Reply:
@clacroix12 this is going to be costly if we keep doing create/delete resources which other tests can utilize, The more I think we should better let the tests decide this and not framework.

Reply:
I agree that tests should choose, however scope is determined via the fixture decorator so I think we need to provide both class and function scope fixtures to our tests so that they can choose which is appropriate for their use-cases. 

I just didn't want us configuring every one of our fixtures to class scope when that can cause unintended issues where tests conflict with each other. As our collection of tests grows we should really try to keep test cases in as isolated of an environment as possible. I think it's better to create/delete resources for tests a bit more aggressively than we might need to in order to avoid issues like a newly implemented test altering the class scoped resource which causes failures in other test cases.